### ISSUE TITLE
Error: 3 INVALID_ARGUMENT: Failed to parse input instances..

### ISSUE THREAD

**Issue by wololodev** [googleapis/google-cloud-node#4005]
I am encountering an "Error: 3 INVALID_ARGUMENT: Failed to parse input instances.." error when using the @google-cloud/aiplatform NodeJS library. I believe this error is related to the request payload that I am sending.

```javascript
const payload = {
  endpoint: 'projects/x/locations/us-central1/endpoints/x',
  instances: [
    {
      countryCode: 'US',
      distanceUserFromVenue: 16646200,
      baseAID: 'stay22',
      destCountry: 'Australia',
      geo_gap: 0.001,
      destTimezone: 660,
      userTimezone: -300,
      osName: 'Mac OS X',
      provider: 'vrbo',
      product: 'map',
      createdAtHour: 2,
      createdAtDayOfWeek: 'Tue',
      distanceUserBin: 'flyfar',
      geoGapBin: 'venue',
      timeZoneDiff: 960
    }
  ]
}

const clientML = new PredictionServiceClient(clientOptions);
await clientML.predict(payload);
```
I'm getting "Failed to parse input instances."

If I do the curl method it works, same payload as above - but obv that is not the optimal way to do it as I want to do it via this library. 

Thanks

#### Environment details

  - which product (packages/*): @google-cloud/aiplatform
  - OS: Mac OS Ventura 13.1 (22C65)
  - Node.js version: 16.*
  - npm version: 8.*
  - google-cloud-node version: 2.6.1

**Comment by dizcology** [comment]
Hello @wololodev: Thanks for creating the issue!  The instances need to be protobuf.Value messages (https://cloud.google.com/nodejs/docs/reference/asset/latest/asset/protos.google.protobuf.value).  

Alternatively, you could use the library's helpers to convert JSON objects, here is an example: https://github.com/googleapis/nodejs-ai-platform/blob/04f7c858217f1a3ce7b1072c7bf8946d39947532/samples/predict-tabular-classification.js#L52

**Comment by wololodev** [comment]
thanks @dizcology that solved it, thanks for referring me to a working example

**Comment by meredithslota** [comment]
@anthowm Can you share more details about what was hard to follow? For v1, we are using a new method of sample authoring — it's intended to more thoroughly cover the API but there may still be some issues.

**Comment by neekey** [comment]
> Hello @wololodev: Thanks for creating the issue! The instances need to be protobuf.Value messages (https://cloud.google.com/nodejs/docs/reference/asset/latest/asset/protos.google.protobuf.value).
> 
> Alternatively, you could use the library's helpers to convert JSON objects, here is an example: https://github.com/googleapis/nodejs-ai-platform/blob/04f7c858217f1a3ce7b1072c7bf8946d39947532/samples/predict-tabular-classification.js#L52

works for me! 

I think the main block is the `google.protobuf.IValue` data structure that makes it difficult to crack.

> Google's Protocol Buffers, which is a binary serialization format developed by Google. It's typically used in gRPC, a high-performance, open-source universal RPC framework.

For anyone who is stuck here, here is the example (we use Vertex AI to serve our object detection model) how you transform your instances data and how you need to transform your response as well:

```js
import vertexAI, { helpers } from '@google-cloud/aiplatform';
export const analyseImage = async (base64Image: string) => {
  // Setup AI Platform Prediction Client
  // https://cloud.google.com/nodejs/docs/reference/aiplatform/latest
  const predictionClient = new vertexAI.PredictionServiceClient({
    apiEndpoint: 'xxxxx-aiplatform.googleapis.com',
  });
  const endpoint = predictionClient.projectLocationEndpointPath(
    'xxxx',
    'xxxx',
    'xxxx',
  );
  const [response] = await predictionClient.predict({
    endpoint: endpoint,
    // https://github.com/googleapis/nodejs-ai-platform/blob/04f7c858217f1a3ce7b1072c7bf8946d39947532/samples/predict-tabular-classification.js#L52
    instances: [helpers.toValue({ image: base64Image })],
  });
  return helpers.fromValue(response.predictions[0] as unknown);
}
```

Notice you will need to use `helpers.toValue` and `helpers.fromValue` to transform the instances and the response.

**Comment by neekey** [comment]
@meredithslota I feel the same about https://github.com/googleapis/google-cloud-node/blob/main/packages/google-cloud-aiplatform/samples/generated/v1/prediction_service.predict.js , which i have came across as well, the main issue is this example give not details about the instance:

```
async function callPredict() {
    // Construct request
    const request = {
      endpoint,
      instances,
    };

    // Run request
    const response = await aiplatformClient.predict(request);
    console.log(response);
  }
```
still this `google.protobuf.IValue` came as a main surprise to me, as none of the thing on the GCP Vertex console mentioned about this:
<img width="1198" alt="image" src="https://github.com/googleapis/google-cloud-node/assets/499870/859ad36d-b44d-4f40-887f-6e73599c3a83">
<img width="1124" alt="image" src="https://github.com/googleapis/google-cloud-node/assets/499870/9a064ba5-ab08-49c2-91a5-c058855afca0">

I think from DX perspective, being predictable, consistent is very important

**Comment by meredithslota** [comment]
Hi @neekey Thank you for the follow-up. I'll route this feedback internally. The examples given in the console are typically "getting started" kinds of examples, and may not represent all use cases. But if you follow the link to the Python example (https://github.com/googleapis/python-aiplatform/blob/main/samples/snippets/prediction_service/predict_custom_trained_model_sample.py) it does mention `google.protobuf.Value` representations.
